itest/lnd_open_channel_test.go

```diff
@@ -967,3 +967,56 @@ func testOpenChannelLockedBalance(ht *lntest.HarnessTest) {
 	// Finally, we check to make sure the balance is unlocked again.
 	ht.AssertWalletLockedBalance(alice, 0)
 }
+
+// testFundingManagerFundingTimeout tests that after an OpenChannel, and before
+// the funding transaction is confirmed, if a user is not the channel initiator,
+// the channel is forgotten after waitBlocksForFundingConf.
+func testFundingManagerFundingTimeout(ht *lntest.HarnessTest) {
+	// Set the maximum wait blocks for funding confirmation.
+	waitBlocksForFundingConf := 10
+
+	// Create nodes for testing, ensuring Alice has sufficient initial
+	// funds.
+	alice := ht.NewNodeWithCoins("Alice", nil)
+	bob := ht.NewNode("Bob", nil)
+
+	// Restart Bob with the custom configuration for funding confirmation
+	// timeout.
+	ht.RestartNodeWithExtraArgs(bob, []string{
+		"--dev.maxwaitnumblocksfundingconf=10",
+	})
+
+	// Ensure Alice and Bob are connected.
+	ht.EnsureConnected(alice, bob)
+
+	// Open the channel between Alice and Bob. This runs through the process
+	// up until the funding transaction is broadcasted.
+	ht.OpenChannelAssertPending(alice, bob,
+		lntest.OpenChannelParams{
+			Amt:     500000,
+			PushAmt: 0,
+		})
```

Comment:
nit: inconsistent formatting. Should either be:
```
	ht.OpenChannelAssertPending(alice, bob, lntest.OpenChannelParams{
		Amt:     500000,
		PushAmt: 0,
	})
```
or
```
	ht.OpenChannelAssertPending(
		alice, bob, lntest.OpenChannelParams{
			Amt:     500000,
			PushAmt: 0,
		},
	)
```

depending on what fits with the line length limit.